genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaClusterServiceImpl.java

```diff
@@ -318,6 +318,68 @@ public void removeAllConfigsForCluster(
      * {@inheritDoc}
      */
     @Override
+    public void addDependenciesForCluster(
+        @NotBlank(message = "No cluster id entered. Unable to add dependencies.")
+        final String id,
+        @NotEmpty(message = "No dependencies entered. Unable to add dependencies.")
+        final Set<String> dependencies
+    ) throws GenieException {
+        this.findCluster(id).getDependencies().addAll(dependencies);
```

Comment:
One thing to be careful about here is that transactions at the Spring level are lost once you make a method call within the same class. By that I mean you do NOT get a new transaction when you call `this.findCluster` since it doesn't go back through the class proxy. In this case I don't think it matters since you're not expecting the other method to commit any intermediate work just something I thought you'd want to know cause we've been burned by it before